src/backend/gl/src/device.rs

```diff
@@ -1627,8 +1627,31 @@ impl d::Device<B> for Device {
         I: IntoIterator,
         I::Item: Borrow<pso::DescriptorSetCopy<'a, B>>,
     {
-        for _copy in copies {
-            unimplemented!()
+        for copy in copies {
+            let copy = copy.borrow();
+
+            let src_set = &copy.src_set;
+            let dst_set = &copy.dst_set;
+            if std::ptr::eq(src_set, dst_set) {
+                panic!("copying within same descriptor set is not currently supported");
+            }
```

Comment:
The Vulkan API allows these in theory (as long as the ranges do not overlap), however the `gfx-hal` docs do not specify whether this is permitted.